tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -368,6 +368,20 @@ class DxcCompiler : public IDxcCompiler2,
         goto Cleanup;
       }
 
+      // Should we fail on this combination?
```

Comment:
Note: We discussed this validation, but I'm not sure I want to add it.  This section needs to be enabled or deleted, once decided.

The problem with the validation is that it means you can't just have something ready to accept debug output when it's available based on the options.  You have to have another intermediate IDxcBlob** variable that either points to your output or not (or ternary operator, or multiple call sites), depending on one or more of the arguments passed in.  Adding this validation will make it a little bit harder to use.